contracts/src/SuccinctGateway.sol

```diff
@@ -282,16 +298,32 @@ contract SuccinctGateway is ISuccinctGateway, FunctionRegistry, TimelockedUpgrad
 
     /// @notice Add the specified prover.
     /// @param _prover The address of the prover to add.
+    function addCustomProver(bytes32 functionId, address _prover) external {
```

Comment:
It would be better to put this function in FunctionRegistry. Would need to move the mapping into there but thats fine.